Rename: ColorSelect를 공용으로 사용할 수 있도록 공용 components 폴더로 옮김

Vegatality opened the pull request:
## #️⃣연관된 이슈

> ex) #이슈번호, #이슈번호

## 📝작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)
- [x] [Rename: ColorSelect를 공용으로 사용할 수 있도록 components 폴더로 옮기고 가져오기 경로를 변경](https://github.com/GihoKo/taskify_team8/commit/2ea247642e504355004c676a7c4bd590ae7582f1)
- [ ] 
- [ ] 

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

Vegatality committed:
Rename: ColorSelect를 공용으로 사용할 수 있도록 components 폴더로 옮기고 가져오기 경로를 변경

Vegatality committed:
Style: sidebar를 dashboard 페이지에 합치면서 스타일 조정

INKmin9 reviewed (approved):
기호님도 쓰시는 것 같길래 후딱 읽어봤습니다..! 깔끔한 코드에 친절한 설명까지.. 멋지십니다👍

Vegatality committed:
Feat: 공용 colorChip 컴포넌트에 선택된 색상을 인자로 전달 받을 수 있도록 추가

Vegatality committed:
Feat: dashboard 및 dashboard 수정 관련 api만 일단 전부 추가

close #77

Vegatality committed:
Feat: tanstack query 세팅 및 tanstack 관련 util 세팅

close #78

Vegatality committed:
Feat: dashboard edit 페이지에서 tanstack query 사용

tanstack query를 페이지에 적용시킨 사례입니다.
참고하셔요.

Vegatality committed:
Fix: 배포 단계에서 edit 페이지 에러 발생한 부분 수정 시도 1

1. window 체크하는 로직을 axios interceptor에 추가
2. 로컬에서 에러는 뜨지 않지만 배포에서 에러가 발생한 파일 추적해서 일단 수정.
(dashboard edit 페이지랑 관련이 있는 것을 추정됩니다.)
3. tanstackquery provider를 react에서 제공해주는 cache라는 훅을 사용하지 않고 직접 만든 함수로 변경(getQueryClient.ts)

Dev-Duke-Seo committed:
Merge branch 'dev' into feature/dashboard-edit

Dev-Duke-Seo commented:
컬러칩 하나에 이리 많은 수고가 깃들었다니 ㅠㅠ 고생많으셨습니다
